examples/browser-webrtc/src/main.rs

```diff
@@ -28,19 +27,22 @@ async fn main() -> anyhow::Result<()> {
         .parse_default_env()
         .init();
 
-    let id_keys = identity::Keypair::generate_ed25519();
-    let local_peer_id = id_keys.public().to_peer_id();
-    let transport = webrtc::tokio::Transport::new(
-        id_keys,
-        webrtc::tokio::Certificate::generate(&mut thread_rng())?,
-    )
-    .map(|(peer_id, conn), _| (peer_id, StreamMuxerBox::new(conn)))
-    .boxed();
-
-    let mut swarm =
-        SwarmBuilder::with_tokio_executor(transport, ping::Behaviour::default(), local_peer_id)
-            .idle_connection_timeout(Duration::from_secs(30)) // Allows us to observe the pings.
-            .build();
+    let mut swarm = libp2p::SwarmBuilder::with_new_identity()
+        .with_tokio()
+        .with_other_transport(|id_keys| {
+            Ok(webrtc::tokio::Transport::new(
+                id_keys.clone(),
+                webrtc::tokio::Certificate::generate(&mut thread_rng())?,
+            )
+            .map(|(peer_id, conn), _| (peer_id, StreamMuxerBox::new(conn))))
+        })?
+        .with_behaviour(|_| ping::Behaviour::default())?
+        .with_swarm_config(
+            libp2p::swarm::Config::with_tokio_executor().with_idle_connection_timeout(
```

Comment:
Should we have `with_swarm_config` take a closure that already hands you to the default instance of `swarm::Config`? I am worried that users need to now align their usage of `with_tokio_executor` here with the above `with_tokio`.

Reply:
Good idea @thomaseizinger. Done in 1fd5a5f6d.